Cython/Compiler/Code.py

```diff
@@ -2083,7 +2083,7 @@ def put_goto(self, lbl):
         self.funcstate.use_label(lbl)
         self.putln("goto %s;" % lbl)
 
-    def put_var_declaration(self, entry, storage_class="",
+    def put_var_declaration(self, entry, function_code, storage_class="",
```

Comment:
I'm not hugely keen on this. The need for a `function_code` parameter changes quite a simple can to quite a complex one.

I think it'd be better to write a separate function `put_unused_if_needed` and run two loops - one to generate the declarations and a second one to generate the `CYTHON_UNUSED_VAR` bits.

It might be more code but I think it'd be simpler code

Reply:
@da-woods Did you mean this the following way: https://github.com/cython/cython/pull/4693/commits/f2195499d853bfc8b63f028ae462105286fbc475?

Reply:
Yes that looks like what I was after.

I had a quick look through and it looked good. I'll have a more detailed look through in the next few days.